src/test/kotlin/uk/gov/dluhc/emsintegrationapi/rest/postal/GetPostalVoteApplicationsIntegrationTest.kt

```diff
@@ -0,0 +1,292 @@
+import uk.gov.dluhc.emsintegrationapi.config.ApiProperties
+import uk.gov.dluhc.emsintegrationapi.config.ERO_CERTIFICATE_MAPPING_CACHE
+import uk.gov.dluhc.emsintegrationapi.config.ERO_GSS_CODE_BY_ERO_ID_CACHE
+import uk.gov.dluhc.emsintegrationapi.config.IntegrationTest
+import uk.gov.dluhc.emsintegrationapi.database.repository.PostalVoteApplicationRepository
+import uk.gov.dluhc.emsintegrationapi.testsupport.ClearDownUtils
+import uk.gov.dluhc.emsintegrationapi.testsupport.WiremockService
+import uk.gov.dluhc.emsintegrationapi.testsupport.testdata.SIGNATURE_WAIVER_REASON
+import uk.gov.dluhc.emsintegrationapi.testsupport.testhelpers.PostalIntegrationTestHelpers
+
+class GetPostalVoteApplicationsIntegrationTest : IntegrationTest() {
+    @Autowired
+    private lateinit var cacheManager: CacheManager
+
+    @Autowired
+    private lateinit var wireMockService: WiremockService
+
+    @Autowired
+    private lateinit var webClient: WebTestClient
+
+    @Autowired
+    private lateinit var apiProperties: ApiProperties
+
+    @Autowired
+    private lateinit var postalVoteApplicationRepository: PostalVoteApplicationRepository
+
+    @Autowired
+    private lateinit var queueMessagingTemplate: QueueMessagingTemplate
+
+    private var apiClient: ApiClient? = null
+
+    private var testHelpers: PostalIntegrationTestHelpers? = null
+
+    private val acceptedPath = "/postalvotes"
+
+    @BeforeEach
+    fun setup() {
+        cacheManager.getCache(ERO_CERTIFICATE_MAPPING_CACHE)?.clear()
+        cacheManager.getCache(ERO_GSS_CODE_BY_ERO_ID_CACHE)?.clear()
+        ClearDownUtils.clearDownRecords(postalRepository = postalVoteApplicationRepository)
+        apiClient = ApiClient(webClient, apiProperties)
+        testHelpers =
+            PostalIntegrationTestHelpers(
+                wiremockService = wireMockService,
+                postalVoteApplicationRepository = postalVoteApplicationRepository,
+                queueMessagingTemplate = queueMessagingTemplate,
+            )
+        testHelpers!!.givenEroIdAndGssCodesMapped()
+    }
+
+    @Test
+    fun `System returns http status 403 if certificate serial number is not attached to the request`() {
+        // When I send a get postal vote applications request without a certificate serial number in the request header
+        val responseSpec = apiClient!!.get(acceptedPath, attachSerialNumber = false)
+
+        // Then I received the http status 403
+        responseSpec.expectStatus().isForbidden
+    }
+
+    @Test
+    fun `System rejects the request with status code 400 if the page size is greater than the configured page size 50`() {
+        // When I send a get postal vote applications request with the page size 51 and the certificate serial number "1234567891"
+        val responseSpec =
+            testHelpers!!.sendGetRequestWithCertificateSerialNumberAndOptionalPageSize(
+                apiClient!!,
+                acceptedPath,
+                "1234567891",
```

Comment:
Similarly, extracting this might be a little nicer?

Reply:
Addressed in latest push.